src/app/core/g3wobject.js

```diff
@@ -73,20 +73,23 @@ proto.un = function(setter, key) {
   type=sync|async
 */
 proto._onsetter = function(when, setter, listener, async, priority=0, once=false) {
-  const settersListeners = this.settersListeners[when];
-  const listenerKey = `${Math.floor(Math.random()*1000000) + Date.now()}`;
+  let listenerKey;
+  const settersListeners = this.settersListeners[when] || {};
   const settersListeneres = settersListeners[setter];
```

Comment:
@volterra79 `settersListeners` vs `settersListeneres` let's find out a different naming...

Reply:
@Raruto right. I'll refactor this part to be more readable